### PR TITLE
OS-8721 Support "server-language"

### PR DESCRIPTION
**Pull request by richlanc**
Support setting active language identifier
seperately from keyboard layout.

### CODE REVIEW

File: src/protocols/rdp/settings.c [hunk 1]
```diff
@@ -269,6 +270,12 @@ enum RDP_ARGS_IDX {
      */
     IDX_SERVER_LAYOUT,
 
+    /**
+     * code_page representing the active language identifier.
+     * See [MS-RDPBCDGR] 2.2.1.11.1.1
```
Comment (SpeeddY) [inline src/protocols/rdp/settings.c:275]: I am looking at `[MS_RDPBCDGR]` 
https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/5073f4ed-1e93-45e1-b039-6e30c385867c

How do I get to `2.2.1.11.1.1` ? Is that a version or a section bookmark ?

**Review by SpeeddY — APPROVED** [review]
Looks fine

File: src/protocols/rdp/settings.c (hunk 1 above)
Comment (richlanc) [reply src/protocols/rdp/settings.c:275]: It's how you navigate the sections.

```
2. -> Messages
2.2 -> Message Syntax
2.2.1 -> Connection Sequence
etc.
```

File: src/protocols/rdp/settings.c [hunk 2]
```diff
@@ -1403,6 +1415,8 @@ void guac_rdp_push_settings(guac_client* client,
     rdp_settings->DesktopHeight = guac_settings->height;
     rdp_settings->AlternateShell = guac_strdup(guac_settings->initial_program);
     rdp_settings->KeyboardLayout = guac_settings->server_layout->freerdp_keyboard_layout;
+    rdp_settings->KeyboardCodePage = guac_settings->server_language;
```
Comment (SpeeddY) [inline src/protocols/rdp/settings.c:1418]: @richlanc 

How did you match the rdp locale codes found in [freerdp](https://pub.freerdp.com/api/locale_8h.html) that we are sending as the `server_language` parameter to this `KeyboardCodePage` parameter?

The documentation on the code page says:

> A 32-bit, unsigned integer. If the flags field does not contain the INFO_UNICODE flag (0x00000010), then this field MUST contain the ANSI code page descriptor being used by the client (for a list of code pages, see [[MSDN-CP]](https://go.microsoft.com/fwlink/?LinkId=89981)) to encode the character fields in the Info Packet and Extended Info Packet (section [2.2.1.11.1.1.1](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/05ada9e4-a468-494b-8694-eb806a0ecc89)). However, if the flags field contains the INFO_UNICODE flag, then the CodePage field MUST contain the active language identifier in the low-word[<16>](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/cbe1ed0a-d320-4ea5-be5a-f2eb6e032853#Appendix_A_16) (for a list of language identifiers, see [[MSDN-MUI]](https://go.microsoft.com/fwlink/?LinkId=90048));

So I presume we are looking at the latter section and we can be sure that guac-server will handle this INFO_UNICODE when it interfaces with rdp.

So going to the link for the active language  identifier in low word: 

> [<16> Section 2.2.1.11.1.1](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/732394f5-e2b5-4ac5-8a0a-35345386b0d1#Appendix_A_Target_16): The Microsoft RDP 6.0 client incorrectly sends the active input locale identifier in the CodePage field of the Info Packet structure.

I am not sure what to make of it?


Furthermore If I go to the 

> for a list of language identifiers, see [[MSDN-MUI]](https://go.microsoft.com/fwlink/?LinkId=90048))

I find a banner saying that language identifiers are deperecated and discouraged?

What am I doing wrong?

**Pull request merged**